targets/TARGET_TOSHIBA/TARGET_TMPM46B/trng_api.c

```diff
@@ -0,0 +1,97 @@
+/* mbed Microcontroller Library
+ * (C)Copyright TOSHIBA ELECTRONIC DEVICES & STORAGE CORPORATION 2017 All rights reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "mbed_error.h"
+#include "tmpm46b_esg.h"
+#include "trng_api.h"
+
+static Result ESG_Config(void)
+{
+    Result ret = ERROR;
+    uint16_t Fintming = 560U;
+
+    // Confirm the ESG core stops
+    if (ESG_GetCalculationStatus() == ESG_CALCULATION_COMPLETE) {
+        // Confirm no interrupt generation
+        if (ESG_GetIntStatus() == DISABLE) {
+            // Set the latch timing & output timing
+            if ((ESG_SetLatchTiming(ESG_LATCH_TIMING_1) == SUCCESS) &&
+                (ESG_SetFintiming(Fintming) == SUCCESS)) {
+                ret = SUCCESS;
+            }
+        }
+    }
+    return ret;
+}
+
+void trng_init(trng_t *obj)
+{
+    // Enable clock supply
```

Comment:
Please add a check that `obj->trng_init == false` when `trng_init` is called.

Reply:
Actually if this function is suppose to initialise the ```obj```. Then it can't trust it's contents to determine if already initialised. If it is required for the driver to know the initialised state. Then it has to save that state itself. How to do that, is implementation dependent.

Reply:
Thanks @mazimkhan, you're absolutely right - @ganesh-ramachandran please ignore this suggestion.